modules/layers/src/path-layer/path-layer.js

```diff
@@ -125,13 +126,15 @@ export default class PathLayer extends Layer {
     const geometryChanged =
       changeFlags.dataChanged ||
       props.fp64 !== oldProps.fp64 ||
+      props.is3D !== oldProps.is3D ||
       (changeFlags.updateTriggersChanged &&
         (changeFlags.updateTriggersChanged.all || changeFlags.updateTriggersChanged.getPath));
 
     if (geometryChanged) {
       this.state.pathTesselator.updateGeometry({
         data: props.data,
         getGeometry: props.getPath,
+        stride: props.is3D ? 3 : 2,
```

Comment:
In my eyes this is `size` or `positionSize`, not `stride`. Stride is how much is skipped between successive elements (usually because other things like colors etc are interleaved).